### PR TITLE
Feature info chart fixes.

### PR DESCRIPTION
**Pull request by na9da**
### What this PR does

Alternate approach to https://github.com/TerriaJS/terriajs/pull/6960 for making y-column of feature info charts configurable.

This PR
-  allows `y-columns` to be used with `<chart>` component in feature info templates without changing the merge strategy of chart traits. It does it by passing the `y-column` as a prop to the `FeatureInfoPanelChart`.
- TSifies `FeatureInfoPanelChart` and purges some `scss` it depends on.
- Default label on x-axis will now say `${y-column-name}(${y-units}) x ${x-column-name}(${x-units})`

### Test me

- Same behaviour as main when a [y-column](https://gist.github.com/steve9164/fa3a23c39ad8c351eab7604f83ec4e7f/#file-test-fi-chart-json-L10) is specified. 
  - Feature info panel for [this branch](http://ci.terria.io/chart-fixes/#share=s-4yZPSBU4peDiyZqMWNAYW23hS9Y&clean&https://gist.githubusercontent.com/steve9164/fa3a23c39ad8c351eab7604f83ec4e7f/raw/d38747e720513d7f9e61c0d736d48eecb1396666/test-fi-chart.json) should be the same as [main](http://ci.terria.io/main/#share=s-4yZPSBU4peDiyZqMWNAYW23hS9Y&clean&https://gist.githubusercontent.com/steve9164/fa3a23c39ad8c351eab7604f83ec4e7f/raw/d38747e720513d7f9e61c0d736d48eecb1396666/test-fi-chart.json).
- Same behaviour as main when [no y-column](https://gist.github.com/na9da/b7df5e7e953aef2c91605e7ee9b69f30#file-chart-json-L28) is specified
  - Feature info panel for [this branch](http://ci.terria.io/chart-fixes/#share=s-gL6i94XLXA2m2lq3PySPh6NmI5s&clean&https://gist.githubusercontent.com/na9da/b7df5e7e953aef2c91605e7ee9b69f30/raw/2df44624bbb94ea12ffe0d3cdf30eb3aff314e6a/chart.json) should be the same as [main](http://ci.terria.io/main/#share=s-gL6i94XLXA2m2lq3PySPh6NmI5s&clean&https://gist.githubusercontent.com/na9da/b7df5e7e953aef2c91605e7ee9b69f30/raw/2df44624bbb94ea12ffe0d3cdf30eb3aff314e6a/chart.json)
- [DEA coastlines chart](http://ci.terria.io/chart-fixes/#clean&https://raw.githubusercontent.com/GeoscienceAustralia/dea-config/master/prod/terria/dea-maps-v8.json&start=%7B%22version%22%3A%228.0.0%22%2C%22initSources%22%3A%5B%7B%22stratum%22%3A%22user%22%2C%22models%22%3A%7B%22D7zEZL%22%3A%7B%22isOpen%22%3Atrue%2C%22knownContainerUniqueIds%22%3A%5B%22%2F%22%5D%2C%22type%22%3A%22group%22%7D%2C%22dkjsia%22%3A%7B%22isOpen%22%3Atrue%2C%22knownContainerUniqueIds%22%3A%5B%22D7zEZL%22%5D%2C%22type%22%3A%22group%22%7D%2C%22dea_coastlines%22%3A%7B%22knownContainerUniqueIds%22%3A%5B%22dkjsia%22%5D%2C%22type%22%3A%22wms%22%7D%2C%22%2F%22%3A%7B%22type%22%3A%22group%22%7D%7D%2C%22workbench%22%3A%5B%22dea_coastlines%22%5D%2C%22timeline%22%3A%5B%22dea_coastlines%22%5D%2C%22initialCamera%22%3A%7B%22west%22%3A137.70378405311394%2C%22south%22%3A-19.126010093894667%2C%22east%22%3A144.41438560256123%2C%22north%22%3A-14.612419369903673%2C%22position%22%3A%7B%22x%22%3A-5203839.909747628%2C%22y%22%3A4205108.579845269%2C%22z%22%3A-2019961.8857561084%7D%2C%22direction%22%3A%7B%22x%22%3A0.7441782485890794%2C%22y%22%3A-0.6013540755191926%2C%22z%22%3A0.29081267197862526%7D%2C%22up%22%3A%7B%22x%22%3A-0.22619250231924448%2C%22y%22%3A0.18278118633466114%2C%22z%22%3A0.9567800111920465%7D%7D%2C%22homeCamera%22%3A%7B%22west%22%3A109%2C%22south%22%3A-45%2C%22east%22%3A158%2C%22north%22%3A-8%7D%2C%22viewerMode%22%3A%223d%22%2C%22showSplitter%22%3Afalse%2C%22splitPosition%22%3A0.4999%2C%22settings%22%3A%7B%22baseMaximumScreenSpaceError%22%3A1%2C%22useNativeResolution%22%3Afalse%2C%22alwaysShowTimeline%22%3Afalse%2C%22baseMapId%22%3A%22basemap-natural-earth-II%22%2C%22terrainSplitDirection%22%3A0%2C%22depthTestAgainstTerrainEnabled%22%3Afalse%7D%2C%22pickedFeatures%22%3A%7B%22providerCoords%22%3A%7B%22https%3A%2F%2Fgeoserver.dea.ga.gov.au%2Fgeoserver%2Fdea%2Fwms%22%3A%7B%22x%22%3A457%2C%22y%22%3A279%2C%22level%22%3A9%7D%2C%22https%3A%2F%2Fstorage.googleapis.com%2Fterria-datasets-public%2Fbasemaps%2Fnatural-earth-tiles%2F%7Bz%7D%2F%7Bx%7D%2F%7BreverseY%7D.png%22%3A%7B%22x%22%3A114%2C%22y%22%3A69%2C%22level%22%3A7%7D%7D%2C%22pickCoords%22%3A%7B%22lat%22%3A-16.261996094297682%2C%22lng%22%3A141.35030359454873%2C%22height%22%3A-10.647001914646191%7D%2C%22current%22%3A%7B%22hash%22%3A26545359815%7D%2C%22entities%22%3A%5B%5D%7D%2C%22stories%22%3A%5B%5D%7D%5D%7D) 
    - Check if the feature info panel chart is rendered correctly, can be expanded, downloaded and toggled from the workbench chart selector list
- [DEA water bodies chart](http://ci.terria.io/chart-fixes/#clean&https://raw.githubusercontent.com/GeoscienceAustralia/dea-config/master/prod/terria/dea-maps-v8.json&start=%7B%22version%22%3A%228.0.0%22%2C%22initSources%22%3A%5B%7B%22stratum%22%3A%22user%22%2C%22models%22%3A%7B%22dn823X%22%3A%7B%22isOpen%22%3Atrue%2C%22knownContainerUniqueIds%22%3A%5B%22%2F%22%5D%2C%22type%22%3A%22group%22%7D%2C%22hNyAH23%22%3A%7B%22isOpen%22%3Atrue%2C%22knownContainerUniqueIds%22%3A%5B%22dn823X%22%5D%2C%22type%22%3A%22group%22%7D%2C%22AS1fd43i505r3%22%3A%7B%22knownContainerUniqueIds%22%3A%5B%22hNyAH23%22%5D%2C%22type%22%3A%22wms%22%7D%2C%22AS1fd43i505r3%3AWaterbody+r4ctum36x_v3%3Ahttps%3A%2F%2Fdata.dea.ga.gov.au%2Fderivative%2Fdea_waterbodies%2F3-0-0%2Ftimeseries%2Fr4ct%2Fr4ctum36x_v3.csv%22%3A%7B%22url%22%3A%22https%3A%2F%2Fdata.dea.ga.gov.au%2Fderivative%2Fdea_waterbodies%2F3-0-0%2Ftimeseries%2Fr4ct%2Fr4ctum36x_v3.csv%22%2C%22chartType%22%3A%22lineAndPoint%22%2C%22chartDisclaimer%22%3A%22Digital+Earth+Australia+Waterbodies+shows+the+wet+surface+area+of+waterbodies+as+estimated+from+satellites.+It+does+not+show+depth%2C+volume%2C+purpose+of+the+waterbody%2C+nor+the+source+of+the+water.+%3Cbr%3EFor+more+detailed+information+see+%3Ca+href%3D%27https%3A%2F%2Fknowledge.dea.ga.gov.au%2Fdata%2Fproduct%2Fdea-waterbodies-landsat%2F%3Ftab%3Doverview%27%3Ehttps%3A%2F%2Fknowledge.dea.ga.gov.au%2Fdata%2Fproduct%2Fdea-waterbodies-landsat%2F%3C%2Fa%3E%22%2C%22name%22%3A%22Waterbody+r4ctum36x_v3%22%2C%22chartPointOnMap%22%3A%7B%22latitude%22%3A-28.11149705034582%2C%22longitude%22%3A137.43607645016837%7D%2C%22styles%22%3A%5B%7B%22id%22%3A%22chart%22%7D%2C%7B%22id%22%3A%22pc_wet%22%2C%22chart%22%3A%7B%22lines%22%3A%5B%7B%22yAxisColumn%22%3A%22px_wet%22%2C%22isSelectedInWorkbench%22%3Afalse%7D%2C%7B%22yAxisColumn%22%3A%22pc_wet%22%2C%22isSelectedInWorkbench%22%3Afalse%7D%5D%7D%7D%5D%2C%22columnTitles%22%3A%5B%22%22%5D%2C%22columnUnits%22%3A%5B%22Date+of+satellite+observation%22%2C%22%25%22%2C%22%23%22%5D%2C%22type%22%3A%22csv%22%7D%2C%22%2F%22%3A%7B%22type%22%3A%22group%22%7D%7D%2C%22workbench%22%3A%5B%22AS1fd43i505r3%22%5D%2C%22timeline%22%3A%5B%5D%2C%22initialCamera%22%3A%7B%22west%22%3A128.81048831700528%2C%22south%22%3A-34.756638051733674%2C%22east%22%3A145.50559981489357%2C%22north%22%3A-25.043326400514292%2C%22position%22%3A%7B%22x%22%3A-4892923.129251108%2C%22y%22%3A4537556.306852739%2C%22z%22%3A-3854206.605548787%7D%2C%22direction%22%3A%7B%22x%22%3A0.6338959314675622%2C%22y%22%3A-0.5878568711867674%2C%22z%22%3A0.5025935207176634%7D%2C%22up%22%3A%7B%22x%22%3A-0.3685177194351761%2C%22y%22%3A0.34175274326576377%2C%22z%22%3A0.8645228469697154%7D%7D%2C%22homeCamera%22%3A%7B%22west%22%3A109%2C%22south%22%3A-45%2C%22east%22%3A158%2C%22north%22%3A-8%7D%2C%22viewerMode%22%3A%223d%22%2C%22showSplitter%22%3Afalse%2C%22splitPosition%22%3A0.4999%2C%22settings%22%3A%7B%22baseMaximumScreenSpaceError%22%3A1%2C%22useNativeResolution%22%3Afalse%2C%22alwaysShowTimeline%22%3Afalse%2C%22baseMapId%22%3A%22basemap-natural-earth-II%22%2C%22terrainSplitDirection%22%3A0%2C%22depthTestAgainstTerrainEnabled%22%3Afalse%7D%2C%22pickedFeatures%22%3A%7B%22providerCoords%22%3A%7B%22https%3A%2F%2Fgeoserver.dea.ga.gov.au%2Fgeoserver%2Fdea%2Fwms%22%3A%7B%22x%22%3A227%2C%22y%22%3A151%2C%22level%22%3A8%7D%2C%22https%3A%2F%2Fstorage.googleapis.com%2Fterria-datasets-public%2Fbasemaps%2Fnatural-earth-tiles%2F%7Bz%7D%2F%7Bx%7D%2F%7BreverseY%7D.png%22%3A%7B%22x%22%3A113%2C%22y%22%3A75%2C%22level%22%3A7%7D%7D%2C%22pickCoords%22%3A%7B%22lat%22%3A-30.857339952390664%2C%22lng%22%3A139.84715488300034%2C%22height%22%3A-204.76778058001491%7D%2C%22current%22%3A%7B%22hash%22%3A800515134%7D%2C%22entities%22%3A%5B%5D%7D%2C%22stories%22%3A%5B%5D%7D%5D%7D)
    - Check if the feature info panel chart is rendered correctly, can be expanded, downloaded and toggled from the workbench chart selector list

### Checklist

- [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
- [ ] I've updated relevant documentation in `doc/`.
- [ ] I've updated CHANGES.md with what I changed.
- [ ] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

**Comment by mwu2018** [comment]
@na9da  I've verified the above testing examples. Also have verified with my own datasets. They all work nicely.

File: lib/ModelMixins/TableMixin.ts [hunk 1]
```diff
@@ -376,14 +376,16 @@ function TableMixin<T extends AbstractConstructor<BaseType>>(Base: T) {
...
       };
 
       return filterOutUndefined(
         lines.map((line) => {
+          const yColumnId = line.yAxisColumn;
           const yColumn = this.findColumnByName(line.yAxisColumn);
-          if (yColumn === undefined) {
+          if (yColumnId === undefined || yColumn === undefined) {
             return undefined;
           }
```
Comment (na9da) [inline lib/ModelMixins/TableMixin.ts:390]: Same as before but just convinces TS that value is not `undefined` when using `yColumnId` below.

File: lib/ReactViews/Custom/Chart/ChartExpandAndDownloadButtons.tsx
Comment (na9da) [inline lib/ReactViews/Custom/Chart/ChartExpandAndDownloadButtons.tsx:1]: Use styled-components.

File: lib/ReactViews/Custom/ChartCustomComponent.ts [hunk 2]
```diff
@@ -325,9 +330,11 @@ export default abstract class ChartCustomComponent<
       chartElements.push(
         React.createElement(Chart, {
           key: "chart",
-          terria: context.terria,
           item: chartItem,
           xAxisLabel: attrs.previewXLabel,
+          // Currently implementation supports showing only one column in the
+          // feature info panel chart
+          yColumn: attrs.yColumns?.[0],
```
Comment (na9da) [inline lib/ReactViews/Custom/ChartCustomComponent.ts:337]: Pass `y-column` as prop to the `FeatureInfoPanelChart`

**Review by nf-s — CHANGES_REQUESTED** [review]
I feel like adding a `y-column` to `FeatureInfoPanelChart` kind of skips the `TableStyle` system. This behaviour is different from `ChartView`/`ChartPanel` - which uses  `ChartItem.showChartInPanel` to determine which charts should be showing.

Ideally `CsvChartCustomComponent` `y-columns` attribute is used to set `TableChartLineStyleTraits.isSelectedInWorkbench` - and then `FeatureInfoPanelChart` shows the first applicable chart item.

What do you think?

**Comment by nf-s** [comment]
I also wonder if it is worth reverting

- https://github.com/TerriaJS/terriajs/pull/6960

if it is no longer needed here - as this was the only use of `MergeStrategy.TopStratum`. How strata is merged is hairy, and I'm not convinced that it was the best approach - it might be better to visit the problem more holistically later

**Review by nf-s — APPROVED** [review]
After discussion: it is difficult to do what I suggested - so merge this in as short-term solution

**Comment by na9da** [comment]
For more context, the reason why we are not able to implement the recommended fix to use `isSelectedInWorkbench` trait is that `isSelectedInWorkbench` is automatically set to true here for the first column:
https://github.com/TerriaJS/terriajs/blob/b73909e26f7193d9047a1460dcab559f3a3e9336/lib/Table/TableAutomaticStylesStratum.ts#L162-L165

To override that programmatically, we'll need to know what columns exists, but that is not known until the chart catalog item is loaded [here in FeatureInfoPanelChart](https://github.com/TerriaJS/terriajs/blob/e3ad3cbdc84b802412c83369f28ef72b9c616488/lib/ReactViews/Custom/Chart/FeatureInfoPanelChart.tsx#L81). It is easier then, to just pass the chosen `y-column` as a prop.

**Pull request merged**